src/prefect/agent.py

```diff
@@ -151,6 +151,21 @@ async def get_infrastructure(self, flow_run: FlowRun) -> Infrastructure:
             infrastructure_document_id
         )
         infrastructure_block = Block._from_block_document(infra_document)
+
+        for override, value in (deployment.infra_overrides or {}).items():
```

Comment:
We have nice utilities for converting from nested dicts to a flat dict representation that can make merging a little easier. See for example https://github.com/PrefectHQ/prefect/blob/main/src/prefect/orion/schemas/core.py#L521-L530, where we have a list of secret fields expressed as `.`-delimited strings and need to modify the nested dictionary at those locations. We convert everything to a flat dict, quickly overwrite only the affected keys, then convert back to a nested dict. It may be overkill for what you’re doing but doesn’t require parsing the for loops. Up to you?

Reply:
Yea the issue is that this object isn't a dictionary so sometimes it's an `attr` get / set, other times I can index into it. Not saying this implementation is the best, but that's why I did it this way